dp/cloud/python/magma/configuration_controller/response_processor/strategies/response_processing.py

```diff
@@ -374,11 +375,15 @@ def _terminate_all_grants_from_response(response: DBResponse, session: Session)
     ) or response.request.payload.get(CBSD_ID)
     if not cbsd_id:
         return
-    cbsd = session.query(DBCbsd).filter(DBCbsd.cbsd_id == cbsd_id).scalar()
     logger.info(f'Terminating all grants for {cbsd_id=}')
-    session.query(DBGrant).filter(DBGrant.cbsd == cbsd).delete()
-    logger.info(f"Deleting all channels for {cbsd_id=}")
-    session.query(DBChannel).filter(DBChannel.cbsd == cbsd).delete()
+    with session.no_autoflush:
```

Comment:
this looks great, can we maybe move to some top level function?

Reply:
leaving as is as discussed